transformers/shared/utilities.go

```diff
@@ -124,6 +124,37 @@ func InsertFieldWithIlk(diffID, headerID int64, ilk, variableName, query, value
 	return tx.Commit()
 }
 
+func InsertFieldWithIlkAndAddress(diffID, headerID int64, address, ilk, variableName, query, value string, db *postgres.DB) error {
+	tx, txErr := db.Beginx()
+	if txErr != nil {
+		return fmt.Errorf("error beginning transaction: %w", txErr)
+	}
+
+	addressID, addressErr := GetOrCreateAddress(address, db)
```

Comment:
Maybe worth using `GetOrCreateAddressInTransaction` here, and rolling back on failure?

Reply:
👍  - I'm also assuming that eventually these two functions will coalesce into one. Is that fair?

Reply:
Did not know about that but it makes a lot of sense, since this is in a transaction.

Reply:
I ended up just passing the addressID here, because we didn't use this method as often as I originally thought. That meant it used the cached version in the cat storage repository.